server/src/server.rs

```diff
@@ -382,10 +408,17 @@ impl<B, L> Builder<B, L> {
 	/// use jsonrpsee_server::ServerBuilder;
 	///
 	/// // Set the ping interval to 10 seconds.
-	/// let builder = ServerBuilder::default().ping_interval(Duration::from_secs(10));
+	/// let builder = ServerBuilder::default().ping_interval(PingConfig::Periodic(Duration::from_secs(60)));
 	/// ```
-	pub fn ping_interval(mut self, interval: Duration) -> Self {
-		self.settings.ping_interval = interval;
+	pub fn ping_interval(mut self, config: PingConfig) -> Self {
```

Comment:
handle the error here instead of `fn build` below.

```suggestion
	pub fn ping_interval(mut self, config: PingConfig) -> Result<Self, Error> {
```